datahub-web-react/src/app/applications/ManageApplications.tsx

```diff
@@ -0,0 +1,185 @@
+import { Button, PageTitle, Pagination, SearchBar, StructuredPopover } from '@components';
+import React, { useEffect, useMemo, useState } from 'react';
+import styled from 'styled-components';
+
+import ApplicationsTable from '@app/applications/ApplicationsTable';
+import CreateNewApplicationModal from '@app/applications/CreateNewApplicationModal/CreateNewApplicationModal';
+import EmptyApplications from '@app/applications/EmptyApplications';
+import { useUserContext } from '@app/context/useUserContext';
+import { PageContainer } from '@app/govern/structuredProperties/styledComponents';
+import { Message } from '@src/app/shared/Message';
+import { useShowNavBarRedesign } from '@src/app/useShowNavBarRedesign';
+import { useGetSearchResultsForMultipleQuery } from '@src/graphql/search.generated';
+import { EntityType } from '@src/types.generated';
+
+const HeaderContainer = styled.div`
+    display: flex;
+    justify-content: space-between;
+    align-items: center;
+    margin-bottom: 0px;
+`;
+
+const SearchContainer = styled.div`
+    display: flex;
+    align-items: center;
+    margin-bottom: 0px;
+`;
+
+// Simple loading indicator at the top of the page
+const LoadingBar = styled.div`
+    position: fixed;
+    top: 0;
+    left: 0;
+    width: 100%;
+    height: 4px;
+    background-color: #1890ff;
+    z-index: 1000;
+    animation: loading 2s infinite ease-in-out;
+
+    @keyframes loading {
+        0% {
+            opacity: 0.6;
+        }
+        50% {
+            opacity: 1;
+        }
+        100% {
+            opacity: 0.6;
+        }
+    }
+`;
+
+const PAGE_SIZE = 10;
+
+const ManageApplications = () => {
+    const isShowNavBarRedesign = useShowNavBarRedesign();
+    const [searchQuery, setSearchQuery] = useState('');
+    const [currentPage, setCurrentPage] = useState(1);
+    const [debouncedSearchQuery, setDebouncedSearchQuery] = useState('*');
+    const [showCreateApplicationModal, setShowCreateApplicationModal] = useState(false);
+
+    const userContext = useUserContext();
+    const canManageApplications = userContext?.platformPrivileges?.manageApplications;
+
+    // Debounce search query input to reduce unnecessary renders
+    useEffect(() => {
+        const timer = setTimeout(() => {
+            setDebouncedSearchQuery(searchQuery);
+        }, 300);
+
+        return () => clearTimeout(timer);
+    }, [searchQuery]);
```

Comment:
so we can actually use lodash's debounce function here which I'd feel more comfortable with than a homebrewed version. you can see different use cases where we have `debounce` or `useDebounce` in the app